api/src/main/java/org/apache/iceberg/transforms/Dates.java

```diff
@@ -77,7 +77,7 @@ public Type getResultType(Type sourceType) {
     if (pred.op() == NOT_NULL || pred.op() == IS_NULL) {
       return Expressions.predicate(pred.op(), fieldName);
     }
-    return ProjectionUtil.truncateIntegerStrict(fieldName, pred, this);
+    return ProjectionUtil.dateIntegerStrict(fieldName, pred, this);
```

Comment:
Technically the new `truncateIntegerStrict` can be used here as well. Just that all the predicates would have to be `LT`/`GT` instead of `LT_EQ`/`GT_EQ`.

Reply:
If the two are interchangeable, why not use the original truncateIntegerStrict for both?

Reply:
The original can't be used because it's directly changing the partition value by 1, which is not true in the case of truncates as it would depend upon the width and how the calculation is done.